plugins/by-name/clangd-extensions/default.nix

```diff
@@ -76,18 +76,30 @@ lib.nixvim.plugins.mkNeovimPlugin {
 
   extraConfig = cfg: {
     warnings = lib.nixvim.mkWarnings "plugins.clangd-extensions" {
-      when = !config.plugins.lsp.enable;
+      when = !((config.plugins.lsp.enable or false) || (config.lsp.servers.clangd.enable or false));
 
       message = ''
-        You have enabled `clangd-extensions` but not the lsp (`plugins.lsp`).
-        You should set `plugins.lsp.enable = true` to make use of the clangd-extensions' features.
+        You have enabled `clangd-extensions` but not the lsp
+        (`plugins.lsp` or `lsp.servers.clangd`). You should set
+        `plugins.lsp.enable = true` or
+        `lsp.servers.clangd.enable = true` to make use of the
+        clangd-extensions' features.
```

Comment:
Style nit: unlike option descriptions and other docs that use markdown formatting, this text is rendered as-is. Including the newlines.

While wrapping the lines like this looks nice in-code, and is usually encouraged, in this scenario it will result in oddly formatted warnings printed to stderr.

While style is subjective, this text must be formatted exactly as you want it to be shown to the user in stderr. Usually that means one line per sentence, or even one line per paragraph.

Reply:
I forgot about, this, thanks!